api/permissions.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework import permissions
+
+class IsProjectOwner(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.owner == request.user
+
+class IsAssigneeOrProjectOwner(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj): # obj here is a Task
+        if request.method in permissions.SAFE_METHODS:
+            return obj.project.owner == request.user or obj.assignee == request.user or request.user.is_staff
+        return obj.project.owner == request.user or obj.assignee == request.user
+
+class IsTaskAssignee(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj): # obj here is a Task
+        return obj.assignee == request.user
+
+
+class IsWorkLogOwner(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return obj.user == request.user
```

Comment:
@VovkMykhailo , have a look please. Make sure that your changes related to the Users properly adopt these permissions